src/runtime/analytics-service.js

```diff
@@ -137,7 +137,14 @@ export class AnalyticsService {
    * @param {string} transactionId
    */
   setTransactionId(transactionId) {
+    const oldTransactionId = this.context_.getTransactionId();
     this.context_.setTransactionId(transactionId);
+    if (oldTransactionId != null) {
```

Comment:
exit early if someone is setting it to the same value.

Reply:
Done